src/protocols/CHIPProtocols.h

```diff
@@ -0,0 +1,66 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines constant enumerations for all public (or
+ *      common) protocols.
+ *
+ */
+
+#ifndef CHIP_PROTOCOLS_H_
+#define CHIP_PROTOCOLS_H_
+
+#include <core/CHIPVendorIdentifiers.hpp>
+
+/**
+ *   @namespace chip::Protocols
+ *
+ *   @brief
+ *     This namespace includes all interfaces within CHIP for CHIP protocols.
+ */
+
+namespace chip {
+namespace Protocols {
+
+//
+// CHIP Protocol Ids (32-bits max)
+//
+
+enum CHIPProtocolId
+{
+    // Common Protocols
+    //
+    // NOTE: Do not attempt to allocate these values yourself.
```

Comment:
is this perhaps copied over from openweave_core?

Reply:
Yes, but this comments are still valid since those ids need to be defined by spec finally. Right now, CHIP stack has defined the protocol ID for Secure Channel Protocol, other protocols are still under review

Reply:
I'd hope that these protocols are all *above* secure channel, i.e. they are protocols supported for dispatch above secure channel...

Reply:
Protocols are not layered, all protocols run on top of messaging layer, Secure Channel Protocol is defined to handle the security protocols of messaging layer .https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/secure_channel/Secure_Channel_Protocol.adoc

Reply:
I understand that everything save rendezvous and mdns is required to run above SecureChannel.

Reply:
I think you are confused by SecureChannel and SecureChannelProtocol. 

This PR is not about how to implement SecureChannelProtocol since Justin has different opinions on if we should implement Security establishment protocols such as Spake2+/Sigma in the protocol level or inside the secure channel like now. 

I am not changing anything here except update enum key kChipProtocol_SecurePairing to  kChipProtocol_SecurityChannel to align with the spec

       - .SetProtocolID(Protocols::kChipProtocol_SecurePairing);
       + .SetProtocolID(Protocols::kChipProtocol_SecurityChannel);

Reply:
thanks for the chat in slack to catch me up, we're on the same page